### DISCUSSION TITLE
Why not providing `DispatcherQueue` as a property on `IDispatcher`?

### DISCUSSION THREAD

**Discussion by xperiandri** [unoplatform/uno.extensions#2281] [Q&A]
## What would you like to be added:

`IDispatcher.DispatcherQueue`

## Why is this needed:

I need `DispatcherQueue` as input in my Elmish.Uno library.
Now I need to bring `Uno.Extensions.Core.WinUI` package as a dependency just because I cannot get a `DispatcherQueue`

## For which Platform:

- iOS
- MacCatalyst
- Android
- WebAssembly

**Comment by nickrandolph** [comment]
@xperiandri the whole point of the interface is to avoid referencing the concrete type (and specifically so we could historically switch implementation between Dispatch and DispatcherQueue.
With regards to the Core.WinUI reference, wouldn't you need this regardless in order to get the implementation of IDispatcher?

**Reply by xperiandri** [reply]
My library depends on WinUI `DispatcherQueue` only.

> With regards to the Core.WinUI reference, wouldn't you need this regardless in order to get the implementation of IDispatcher?

Yes, but it is not an Uno.Extension
